CMakeLists.txt

```diff
@@ -219,7 +219,11 @@ find_package(ZLIB)
 find_library(SNAPPY snappy)
 
 if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
-  set(CMAKE_PREFIX_PATH "/usr/local/opt/icu4c" ${CMAKE_PREFIX_PATH})
+  if(NOT CMAKE_PREFIX_PATH MATCHES ".*\/icu4c.*")
```

Comment:
IMO we should not add paths that were not created by the CMake script / part of the repo to the prefix path in our CMake script, but I can't find any discussion of best practices around this.

If you want to keep this default, at least append rather than prepend the path; that way it will not override developer-provided paths.